lib/assembly/inventory_repo.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Assembly.InventoryRepo do
+  @moduledoc """
+  Communication to our inventory microservice (or inventory database currently)
+  """
+
+  use Ecto.Repo,
+    otp_app: :assembly,
+    adapter: Ecto.Adapters.MyXQL
+
+  import Ecto.Query
+
+  @doc """
+  Gets a list of all inventory skus in our system and the current available
+  count of parts for that sku.
+  """
+  def get_sku_counts() do
+    query =
+      from(s in "inventory_skus",
+        left_join: p in "inventory_parts", on: s.id == p.sku_id,
+        where: is_nil(p.assembly_build_id),
+        where: is_nil(p.rma_description),
+        where: p.location_id not in ^excluded_picking_locations(),
```

Comment:
This was my concern with having this service read from the DB. This is no longer a build service if it's also having to retain intimate knowledge of how inventory works.

I think we need to abandoned this approach before we go any further otherwise we're doing very little to improve assembly and break it out of HAL.

Reply:
The more I'm thinking about this the more I think we may need to step back and think about this a bit. Kitting alone is a lot for the assembly service to know and make sense of. The assembly service really only needs to know about the availability of a component, not which SKU specifically.

The UI is where those 2 disparate system's data should be married up.

Reply:
@doomspork I asked this in the planning meeting. I said it would be faster to just add a read only database to map this stuff out. It's all contained in this file that can be easily replaced with grpc in the future. It's not even using structs to avoid making decisions on data structure yet.

I don't understand what you say they should be married in the UI. Kitting needs to be stored in some location, and to me assembly seems to make the best sense. Inventory is already large and possibly might grow (depending how accounting goes). The catalog shouldn't care about anything technical (like kitting or skus). Assembly should be the in between. It gets information from the catalog, and pulls information from inventory. At it's max, it would contain kitting logic (the microserice to include / care about it), and build information (id, status, the parts attached to it).

Reply:
We did discuss this but now seeing it in action I am reminded there is a lot of logic to inventory that would not live here in addition to HAL.

Kitting is another term for part interchangeability, that's an inventory function. How can you put a kit together without knowing all of the SKUs in inventory? That knowledge shouldn't need to live in Assembly.

Reply:
Part interchangeablity between the catalog and actual parts though. Now inventory needs to know about the catalog. If we store it in assembly, it can be just fkeys to other services. Assembly doesn't even need to know what those fkeys point to. All of that can be done front end.